src/peerpod-ctrl/controllers/peerpod_controller.go

```diff
@@ -37,8 +37,8 @@ import (
 // PeerPodReconciler reconciles a PeerPod object
 type PeerPodReconciler struct {
 	client.Client
-	Scheme   *runtime.Scheme
-	Provider provider.Provider
+	Scheme    *runtime.Scheme
+	Providers map[string]provider.Provider
```

Comment:
Any thoughts on how the different provider configs will be stored in the same cm/secret?

Reply:
I have been focusing only on the `ibmcloud` and `ibmcloud-powervs` providers which do not have any conflicting env vars. This let's me add all env vars to the same cm/secret.

Also, the set of env vars used by peerpod-ctrl is small since it only makes a call to delete the PodVM at the moment. All it requires is `IBMCLOUD_API_KEY` secret.

If we continue with this approach for the wider [issue](https://github.com/confidential-containers/cloud-api-adaptor/issues/2219), we may need to add prefix to conflicting env vars. I believe that is already the case for many env vars.